star-ros2/src/star_simulation/test/test_stall_detection.py

```
@@ -0,0 +1,148 @@
+        cls.node.destroy_node()
+        rclpy.shutdown()
+
+    def spin_for(self, seconds):
+        end = time.time() + seconds
+        while time.time() < end:
+            rclpy.spin_once(self.node, timeout_sec=0.05)
+
+    def test_stall_triggers_estop(self):
+        """Drive robot into a wall, verify stall detected -> e-stop."""
+        # Spawn a thick wall directly in front of the robot
+        subprocess.run([
```

Ruff (0.15.9)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)